modules/outbound-command-event-handler/test/integration/application/process_bulk_quotes_callback.test.ts

```diff
@@ -0,0 +1,607 @@
+  IProcessPartyInfoCallbackCmdEvtData,
+  IProcessSDKOutboundBulkAcceptPartyInfoCmdEvtData,
+  IProcessSDKOutboundBulkPartyInfoRequestCmdEvtData,
+  IProcessSDKOutboundBulkPartyInfoRequestCompleteCmdEvtData,
+  IProcessSDKOutboundBulkQuotesRequestCmdEvtData,
+  IProcessSDKOutboundBulkRequestCmdEvtData,
+  IRedisBulkTransactionStateRepoOptions,
+  KafkaCommandEventProducer,
+  KafkaDomainEventConsumer,
+  ProcessBulkQuotesCallbackCmdEvt,
+  ProcessPartyInfoCallbackCmdEvt,
+  ProcessSDKOutboundBulkAcceptPartyInfoCmdEvt,
+  ProcessSDKOutboundBulkPartyInfoRequestCmdEvt,
+  ProcessSDKOutboundBulkPartyInfoRequestCompleteCmdEvt,
+  ProcessSDKOutboundBulkQuotesRequestCmdEvt,
+  ProcessSDKOutboundBulkRequestCmdEvt,
+  RedisBulkTransactionStateRepo,
+} from "@mojaloop/sdk-scheme-adapter-private-shared-lib"
+import { randomUUID } from "crypto";
+
+// Tests can timeout in a CI pipeline so giving it leeway
+jest.setTimeout(20000)
+
+const logger: ILogger = new DefaultLogger('bc', 'appName', 'appVersion'); //TODO: parameterize the names here
+const messageTimeout = 2000;
+
+// Setup for Kafka Producer
+const commandEventProducerOptions: IKafkaEventProducerOptions = {
+    brokerList: 'localhost:9092',
+    clientId: 'test-integration_client_id',
+    topic: 'topic-sdk-outbound-command-events'
+}
+const producer = new KafkaCommandEventProducer(commandEventProducerOptions, logger)
+
+// Setup for Kafka Consumer
+const domainEventConsumerOptions: IKafkaEventConsumerOptions = {
+  brokerList: 'localhost:9092',
+  clientId: 'test-integration_client_id',
+  topics: ['topic-sdk-outbound-domain-events'],
+  groupId: "domain_events_consumer_client_id"
+}
+var domainEvents: Array<DomainEvent> = []
+const _messageHandler = async (message: DomainEvent): Promise<void>  => {
+  console.log('Domain Message: ', message);
+  domainEvents.push(message);
+}
+const consumer = new KafkaDomainEventConsumer(_messageHandler.bind(this), domainEventConsumerOptions, logger)
+
+// Setup for Redis access
+const bulkTransactionEntityRepoOptions: IRedisBulkTransactionStateRepoOptions = {
+  connStr: 'redis://localhost:6379'
+}
+const bulkTransactionEntityRepo = new RedisBulkTransactionStateRepo(bulkTransactionEntityRepoOptions, logger);
+
+
+describe("Tests for Outbound Command Event Handler", () => {
+
+  beforeEach(async () => {
+    domainEvents = [];
+  });
+
+  beforeAll(async () => {
+    await producer.init();
+    await consumer.init();
+    await consumer.start();
+    await bulkTransactionEntityRepo.init();
+  });
+
+  afterAll(async () => {
+    await producer.destroy();
+    await consumer.destroy();
+    await bulkTransactionEntityRepo.destroy();
+  });
+
+  test("Given Inbound command event ProcessBulkQuotesCallback for success requests \
+      Then the logic should update the individual batch state to AGREEMENT_PROCESSING, \
+      And create batches based on FSP that has DISCOVERY_ACCEPTED state \
+      And publish BulkQuotesRequested per each batch \
```

Comment:
Published event should be BulkQuotesCallbackProcessed.

Reply:
Fixed description.